### PR TITLE
[iOS] Open Content URL 

### PR DESCRIPTION
**Pull request by ry-itto**
## Issue
- close N/A

## Overview (Required)
- Open content URL
- Open Staff/Contributor URL
- Clean About

## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: ios/Sources/MediaFeature/MediaListFeature.swift [hunk 1]
```diff
@@ -71,8 +72,16 @@ let mediaListReducer = Reducer<MediaListState, MediaListAction, MediaEnvironment
     case .moreDismissed:
         state.moreActiveType = nil
         return .none
-    case .tap(let content):
-        // TODO: open content page
+    case .tap(let id):
+        if let selectedFeedContent = state.feedContents.first(where: { $0.id == id }),
```
Comment (ostk0069) [inline ios/Sources/MediaFeature/MediaListFeature.swift:76]: if we weed to search the feedContent, I think it's better to path the whole `FeedContent` or path the url, not id.

File: ios/Sources/UIApplicationClient/Client.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public protocol UIApplicationClientProtocol {
+    func open(url: URL, options: [UIApplication.OpenExternalURLOptionsKey: Any]) -> AnyPublisher<Bool, Never>
+}
+
+public struct UIApplicationClient: UIApplicationClientProtocol {
+    public init() {}
+
+    public func open(url: URL, options: [UIApplication.OpenExternalURLOptionsKey: Any]) -> AnyPublisher<Bool, Never> {
+        Future<Bool, Never> { promise in
+            UIApplication.shared.open(url, options: options) { success in
```
Comment (ostk0069) [inline ios/Sources/UIApplicationClient/Client.swift:13]: By showing WebContent in Apps, there are many ways to do.
Is this the way android app does?

File: ios/Sources/HomeFeature/HomeListFeature.swift [hunk 3]
```diff
@@ -22,26 +23,38 @@ public struct HomeListState: Equatable {
...
+        self.applicationClient = applicationClient
     }
 }
 
 public let homeListReducer = Reducer<HomeListState, HomeListAction, HomeListEnvironment> { state, action, environment in
     switch action {
-    case .selectFeedContent:
-        // TODO: open content page
+    case let .selectFeedContent(id):
+        if let selectedFeedContent = state.feedContents.first(where: { $0.id == id }),
```
Comment (ostk0069) [inline ios/Sources/HomeFeature/HomeListFeature.swift:49]: ditto

File: ios/Sources/FavoritesFeature/FavoritesListFeature.swift [hunk 4]
```diff
@@ -11,25 +12,37 @@ public struct FavoritesListState: Equatable {
...
+        self.applicationClient = applicationClient
     }
 }
 
 public let favoritesListReducer = Reducer<FavoritesListState, FavoritesListAction, FavoritesListEnvironment> { state, action, environment in
     switch action {
-    case .tap:
-        // TODO: open content page
+    case let .tap(id):
+        if let selectedFeedContent = state.feedContents.first(where: { $0.id == id }),
```
Comment (ostk0069) [inline ios/Sources/FavoritesFeature/FavoritesListFeature.swift:37]: ditto

File: ios/Sources/AboutFeature/StaffCell.swift [hunk 5]
```diff
@@ -1,17 +1,20 @@
 import Component
+import Model
 import Styleguide
 import SwiftUI
 
 public struct StaffCell: View {
     private let staff: Staff
+    private let onTap: (Staff) -> Void
```
Comment (ostk0069) [inline ios/Sources/AboutFeature/StaffCell.swift:8]: I think it's better to unify the name of action. 
In current app we name `tapAction`. ref: https://github.com/DroidKaigi/conference-app-2021/blob/b82b2bb4a64a13400c4754abbb0b6c1f4cecbb4d/ios/Sources/Component/Card/SmallCard.swift#L17

File: ios/Sources/AboutFeature/ContributorCell.swift [hunk 6]
```diff
@@ -1,17 +1,20 @@
 import Component
+import Model
 import Styleguide
 import SwiftUI
 
 public struct ContributorCell: View {
     private let contributor: Contributor
+    private let onTap: (Contributor) -> Void
```
Comment (ostk0069) [inline ios/Sources/AboutFeature/ContributorCell.swift:8]: ditto

File: ios/Sources/AboutFeature/AboutFeature.swift [hunk 7]
```diff
@@ -1,31 +1,68 @@
...
+            return environment.applicationClient.open(
+                url: url,
+                options: [:]
+            )
+            .eraseToEffect()
+            .map(AboutAction.urlOpened)
+        }
+        return .none
+    case let .tapContributor(contributor):
+        if let url = URL(string: contributor.urlString) {
```
Comment (ostk0069) [inline ios/Sources/AboutFeature/AboutFeature.swift:56]: ditto

File: ios/Sources/AboutFeature/AboutFeature.swift [hunk 8]
```diff
@@ -1,31 +1,68 @@
...
+    case .refresh:
+        // TODO: Fetch data from server
+        state.staffs = [.mock(), .mock(), .mock()]
+        state.contributors = [.mock(), .mock(), .mock()]
+        return .none
     case .selectedPicker(let selectedType):
         state.selectedType = selectedType
         return .none
+    case let .tapStaff(staff):
+        if let url = URL(string: staff.urlString) {
```
Comment (ostk0069) [inline ios/Sources/AboutFeature/AboutFeature.swift:46]: ditto

**Comment by ry-itto** [comment]
@ostk0069 
I fixed 🙏🏼 
Please check this again!

**Review by ostk0069 — APPROVED** [review]

**Pull request merged**